src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -229,6 +230,19 @@ private async ValueTask<CancellationTokenRegistration> HandleWriteStartState(Can
                 throw new InvalidOperationException(SR.net_quic_writing_notallowed);
             }
 
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
This should be handled in the `cancellationToken.UnsafeRegister`. Is the await for `_state.SendResettableCompletionSource` a problem? Do we need to throw before any awaits?

cc @CarnaViire

Reply:
The ReadWriteAsync_PrecanceledOperations_ThrowsCancellationException did not pass without the (read) change.
it seems like the pre-check is cheap so I added instead of investigating the state machine and signaling.